src/components/AssetSearchForm/AssetSearchForm.tsx

```diff
@@ -0,0 +1,261 @@
+    <Form.Item
+      {...(index === 0 ? formItemLayout : formItemLayoutWithoutLabel)}
+      label={index === 0 ? metadataLabel : ''}
+      required={false}
+      key={data.id}
+    >
+      <Input.Group compact={true} style={{ display: 'flex' }}>
+        <Input
+          className={`meta-key ${data.id}`}
+          style={{ width: '45%', borderRight: 0 }}
+          placeholder={metadataKey}
+          value={data.key}
+          onPressEnter={pressEnter}
+          onChange={onChangeInput('key', index)}
+        />
+
+        <Input
+          style={{
+            width: '10%',
+            borderLeft: 0,
+            pointerEvents: 'none',
+            backgroundColor: '#fff',
+          }}
+          value=":"
+          disabled={true}
+        />
+        <Input
+          className={`meta-value ${data.id}`}
+          style={{ width: '45%', borderLeft: 0 }}
+          value={data.value}
+          placeholder={metadataValue}
+          onPressEnter={pressEnter}
+          onChange={onChangeInput('value', index)}
+        />
+        {metadata.length > 1 ? (
+          <Icon
+            type="minus-circle-o"
+            style={{
+              position: 'absolute',
+              right: -32,
+              top: '50%',
+              transform: 'translateY(-50%)',
+            }}
+            onClick={() => onRemoveMetadata(data.id)}
+          />
+        ) : null}
+      </Input.Group>
+    </Form.Item>
+  ));
+
+  return (
+    <Form layout="horizontal" onSubmit={submit}>
+      <Form.Item label={nameField} {...formItemLayout}>
+        {getFieldDecorator('name')(
+          <Input
+            className="name"
+            maxLength={50}
+            placeholder={nameField}
+            onPressEnter={pressEnter}
+          />
+        )}
+      </Form.Item>
+      <Form.Item label={descriptionField} {...formItemLayout}>
+        {getFieldDecorator('description')(
+          <Input
+            className="description"
+            maxLength={500}
+            placeholder={descriptionField}
+            onPressEnter={pressEnter}
+          />
+        )}
+      </Form.Item>
+      {metadataFields}
+      <Form.Item {...formItemLayoutWithoutLabel}>
+        <Button
+          htmlType="button"
+          type="dashed"
+          className="add-more-metadata"
+          onClick={onAddMetadata}
+          style={{ width: '100%' }}
+        >
+          <Icon type="plus" style={{ marginRight: 8 }} />
+          {addMetadata}
+        </Button>
+      </Form.Item>
+
+      {onSubmit && (
+        <Form.Item {...formItemLayoutWithoutLabel}>
+          <Button
+            htmlType="button"
+            className="submit"
+            type="primary"
+            onClick={submit}
+            style={{ float: 'right' }}
+          >
+            {search}
+          </Button>
+        </Form.Item>
+      )}
+    </Form>
+  );
+};
+
+const AssetSearchFormHOC = Form.create({
```

Comment:
What does HOC stand for?

Reply:
HOC provide form property with bunch of methods, which helps to grep data from form (such as getFieldDecorator, getFieldValue, setFieldsValue)